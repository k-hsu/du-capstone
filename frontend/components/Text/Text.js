import React from "react";
import styled from "@emotion/styled";
import { typography } from "../../theme";

const StyledText = styled.text`
  ${({ fontFamily }) => `font-family: "${fontFamily}";`}
  ${({ as, lineHeight }) =>
    as || lineHeight
      ? `line-height: ${lineHeight ?? typography[as].lineHeight};`
      : ""}
  ${({ as, fontWeight }) =>
    as || fontWeight
      ? `font-weight: ${fontWeight ?? typography[as].fontWeight};`
      : ""}
  ${({ as, fontSize }) =>
    as || fontSize ? `font-size: ${fontSize ?? typography[as].fontSize};` : ""}
  ${({ fontStyle }) => (fontStyle ? `font-style: ${fontStyle};` : "")}
  ${({ color }) => (color ? `color: ${color};` : "")}
  ${({ textDecoration }) =>
    textDecoration ? `text-decoration: ${textDecoration};` : ""}
  ${({ cursor }) => (cursor ? `cursor: ${cursor};` : "")}
`;

const Text = ({
  children,
  as = "p",
  fontFamily = "IBM Plex Sans",
  ...props
}) => {
  return (
    <StyledText as={as} fontFamily={fontFamily} {...props}>
      {children}
    </StyledText>
  );
};

export default Text;
