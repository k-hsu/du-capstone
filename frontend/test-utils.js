import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
