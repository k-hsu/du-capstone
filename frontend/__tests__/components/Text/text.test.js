import React from "react";
import { render, screen } from "../../../test-utils";
import Text from "../../../components/Text/Text";

describe("Text", () => {
  it("should render text component", () => {
    render(<Text>text mex</Text>);
    expect(screen.getByText("text mex")).toBeInTheDocument();
    expect(screen.getByText("text mex")).toHaveStyle(
      `font-family: "IBM Plex Sans"`
    );
  });
});
