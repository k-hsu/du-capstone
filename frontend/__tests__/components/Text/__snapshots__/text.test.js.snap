// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render default text component styles 1`] = `
<p
  className="css-ms6223"
  fontFamily="IBM Plex Sans"
>
  text mex
</p>
`;

exports[`Text should render h1 component styles 1`] = `
<h1
  className="css-5egnv5"
  fontFamily="IBM Plex Sans"
>
  text mex
</h1>
`;

exports[`Text should render h2 component styles 1`] = `
<h2
  className="css-drfpaa"
  fontFamily="IBM Plex Sans"
>
  text mex
</h2>
`;

exports[`Text should render h3 component styles 1`] = `
<h3
  className="css-1n9884d"
  fontFamily="IBM Plex Sans"
>
  text mex
</h3>
`;
